### PR TITLE
Merge "api doc" and "additional details" features

### PR DESCRIPTION
**Pull request by jotak**
- API doc feature is removed / replaced with more generic "additional details" one
- Add optional icon display with additional details (to keep API doc icons)
- Kiali won't embed documentation anymore (so no more client call to get documentation)
- For "dogfooding", Kiali service & deployment itself have a link to our swagger API

Fixes kiali/kiali#1567 (part-of)
Frontend PR: https://github.com/kiali/kiali-ui/pull/1667

### CODE REVIEW

**Comment by jotak** [comment]
In this PR, I'm adding some annotations to Kiali service & deployment for "dogfooding": even if it's not the most interesting thing to have access to API from within Kiali, still it allows to easily & frequently make sure the feature is not broken.

@jmazzitelli because of that I'm modifying template files, can you have a look / make sure I'm not forgetting something?

**Comment by jmazzitelli** [comment]
Yeah, do not merge this yet. The operator has gone under a major refactoring, and we've duplicated ansible scripts to support multiple versions. I'm in the process of trying to make sure everything is working in master and v1.12 branch. I was hoping there would be no churn in here until everything is done.

**Comment by jotak** [comment]
Ok, the changes in the templates are quite secondary, I could remove them and maybe put them back later

File: operator/roles/default/kiali-deploy/templates/kubernetes/deployment.yaml [hunk 1]
```diff
@@ -6,6 +6,9 @@ metadata:
   labels:
     app: kiali
     version: {{ kiali_vars.deployment.version_label }}
+  annotations:
```
Comment (jmazzitelli) [inline operator/roles/default/kiali-deploy/templates/kubernetes/deployment.yaml]: are you sure you want this here on the deployment itself? Or do you want it on the actual pod? (see template section below)

Comment (jotak) [reply operator/roles/default/kiali-deploy/templates/kubernetes/deployment.yaml]: it's the deployment in this case

File: operator/roles/default/kiali-deploy/templates/openshift/deployment.yaml [hunk 2]
```diff
@@ -6,6 +6,9 @@ metadata:
   labels:
     app: kiali
     version: {{ kiali_vars.deployment.version_label }}
+  annotations:
```
Comment (jmazzitelli) [inline operator/roles/default/kiali-deploy/templates/openshift/deployment.yaml]: are you sure you want these here on the deployment itself? or on the pod (see template section below)

**Comment by jmazzitelli** [comment]
@jotak OK, all the new operator refactoring has been merged into master and v1.12 branches. I'll take a look at this PR tomorrow and give a more thorough review.

But in short, the new operator now supports multiple version streams - so you will now see several sets of independent ansible roles - "default" (which is the latest/master version), "v1.0" (which installs the older v1.0 kiali) and "v1.12" (which installs kiali built out of the v1.12 branch).

So you normally only have to touch the "default" ansible files unless you are planning on backporting this to the v1.12 branch, at which time you need to make similar changes (if required) to the v1.12 ansible files.

**Comment by jotak** [comment]
@jmazzitelli ok. For this PR, there's nothing to backport, so sticking to changing default files only should be good.

File: operator/roles/v1.12/kiali-deploy/defaults/main.yml [hunk 3]
```diff
@@ -13,8 +13,11 @@ kiali_defaults:
   istio_namespace: ""
   version: "v1.12"
 
-  additional_display_details: []
-
+  additional_display_details:
```
Comment (jmazzitelli) [inline operator/roles/v1.12/kiali-deploy/defaults/main.yml]: This file is in `roles/v1.12` - are you sure you want to edit this file?

You mentioned elsewhere that there is no backporting required for this PR. I think you mean to change the defaults/main.yml in the `roles/default` ansible role location.

File: operator/deploy/kiali/kiali_cr.yaml [hunk 4]
```diff
@@ -24,7 +24,8 @@ spec:
 #  additional_display_details: []
 #
 # A list of additional details that Kiali will look for in annotations and display, for every workloads and services, in their respective details pages.
-# Each item in the list is an object with "annotation" and "title" fields to indicate which annotation is to be displayed and how it should be displayed.*
+# Each item in the list is an object with "annotation", "title" and "icon_annotation" fields to indicate which annotation is to be displayed and how it should be displayed.
+# At the moment, "icon_annotation" possible values are "rest", "grpc" and "graphql".
```
Comment (jmazzitelli) [inline operator/deploy/kiali/kiali_cr.yaml]: This is trivial, but I just noticed that this example below doesn't show "icon_annotation" - is this a required field? If so, should probably add it to the example below.

File: operator/roles/v1.12/kiali-deploy/defaults/main.yml (hunk 3 above)
Comment (jotak) [reply operator/roles/v1.12/kiali-deploy/defaults/main.yml]: oops, good catch, I don't know how I came up modifying the wrong one!

File: operator/deploy/kiali/kiali_cr.yaml (hunk 4 above)
Comment (jotak) [reply operator/deploy/kiali/kiali_cr.yaml]: It's an optional field, but I've improved the documentation anyway

File: operator/roles/default/kiali-deploy/defaults/main.yml [hunk 5]
```diff
@@ -13,7 +13,10 @@ kiali_defaults:
   istio_namespace: ""
   version: "default"
 
-  additional_display_details: []
+  additional_display_details:
```
[the remark this answers is not in the document]
Comment (jotak) [reply operator/roles/default/kiali-deploy/defaults/main.yml]: Yes, I understand.. but this default is also mainly for our dogfooding, I mean it's not extremely important if it's lost per user config, I think. But I can mention that in the doc, yes.

File: operator/deploy/kiali/kiali_cr.yaml [hunk 6]
```diff
@@ -21,24 +21,23 @@ spec:
...
+# By default, this settings recognizes API Documention links via annotation "kiali.io/api-spec" and icon-annotation "kiali.io/api-type".
+# For example, it would make Kiali recognize these annotations in a service or a workload definition (Deployment, StatefulSet, etc.) to display the appropriate link and text:
 #    annotations:
-#      my-domain/api-documentation: http://link/to/my/doc
-#      my-domain/api-type: rest
-#      my-domain/gitsha: c6da17cf
+#      kiali.io/api-spec: http://link/to/my/doc
+#      kiali.io/api-type: rest
+# Should you change this setting for your own custom annotations, keep in mind that it would override the current default.
+# So you would have to copy the "API Documentation" setting as shown above if you want to preserve these links.
```
Comment (jotak) [inline operator/deploy/kiali/kiali_cr.yaml]: @jmazzitelli i've reformulated ... better like that?

**Pull request merged**